### ISSUE TITLE
Implicit conversion from null to empty string with PersonName causes problems in tests

### ISSUE THREAD

**Issue by ayedo** [fo-dicom/fo-dicom#1395]
First of all, thank you very much for your work on fo-dicom! I have very much enjoyed using the library so far.  Unfortunately, we are running into some problems in tests because of the following behavior of the library:

**Describe the bug**
`PersonName` has optional parameters `Middle`, `Suffix` and `Prefix` which by default are set to `null`. 
When reading values from a `DicomDataSet` the values which were stored with a `null` value, the method returns the empty string. This behavior seems inconsistent.

**To Reproduce**
run the following xUnit test: 

```
    [Fact]
    public void Person_Name_Middle_Name_Round_Trip()
    {
        string? expectedMiddle = null;
        
        var set = new DicomDataset {new DicomPersonName(DicomTag.PatientName, "First", "Last", Middle: expectedMiddle)};

        var result = set.GetDicomItem<DicomPersonName>(DicomTag.PatientName);

        result.Middle.Should().Be(expectedMiddle); // fails with message: Expected result.Middle to be <null>, but found "".
    }
```

**Expected behavior**
There are different options that I would consider consistent behavior:
- when the optional values were set as `null`,  also return `null` when reading them
- do not make the the `Middle`, `Suffix`, `Prefix` parameters optional, and force the user to set them to the empty string
- set the default value of the `Middle`, `Suffix`, and `Prefix` parameter to the empty string, and throw an `ArgumentNullException` if they are set to null

**Environment**
Fellow Oak DICOM version: 5.0.2
OS: macOS Monterey 12.4
Platform: .net core 6.0.300

**Comment by mrbean-bremen** [comment]
I think the problem stems from the fact that for string values in DICOM, there is no difference between a tag with no values (e.g. VM 0), which is usually represented by `null`, and an empty string. 
However, this case is a bit different as `Middle`, `Suffix` and `Prefix` are only parts of one string, so this reasoning may not apply here. While for a whole string, a `null` value can be set and later retrieved (at least until it is written to a file), this is not possible for string parts - we just have to decide if we want to return `null` or a empty string for an empty string part. Returning an empty string as is probably makes the most sense. So, your first option will not work (at least if we don't want to save each part separately). 

I'm not sure what the best solution is here. I think, setting all values to `null` (the default) could create an empty string (which is indistinguishable from `null` in DICOM), and setting all values to empty strings could generate the string `"^^"`. Not making the parameters optional would be inconvenient and not downwards compatible, so also not a good option IMO. I would say setting the default parameters to an empty string would be the best solution, though I'm not sure if we shall forbid using `null` here - after all, we also allow this for other string values, _and_ it could be a compatibility problem.

I don't see this as a bug, BTW, but I agree that the behavior is a bit inconsistent, and my proposal would be to just change the default values, and leave the code as is (which does not differentiate between `null` and empty strings by using `string.IsNullOrEmpty()`.

**Comment by ayedo** [comment]
Thank you for your quick reply. Your explanation make sense, and I agree that it's important to keep backwards compatibility. I was not sure if it's a bug or intended behaviour, as you said, with DICOM some things are just inconsistently defined in the standard itself. Changing the default values sounds like a good compromise to me, as it adds consistency with the least chance of backwards incompatibility.  I'm new to the project, and I'm not sure how pull-requests work here, but if you guys are open to those, I can make the changes and send you one?

**Comment by mrbean-bremen** [comment]
Of course, go ahead! Pull requests work the same as in any project, and there should be a template with tasks for the PR.

**Comment by ayedo** [comment]
OK, it might take am moment for me to find time for it, but as you said, it's not really a bug, and it's also not urgent.
